pkg/arvo/sys/zuse.hoon

```diff
@@ -4244,32 +4245,82 @@
             [b+8 t+9 n+10 f+12 r+13 ~]
           =*  wow  `(map @t @)`(malt lip)
           (sear ~(get by wow) low)
-        =*  tuf  ;~(pfix (just 'u') (cook tuft qix:ab))
-        ;~(pose doq fas soq bas loo tuf)
+        =*  sup
+          =*  hsb  0xd800
+          =*  hst  0xdbff
+          =*  lsb  0xdc00
+          =*  lst  0xdfff
+          %+  cook
+            |=  [hig=@ low=@]
+              ^-  @t
+              (tuft :(add 0x1.0000 low (lsh [1 5] hig)))
+          ;~  plug
+            ;~  pfix
+              (just 'u')
+              %+  sear
+                |=  sur=@ux
+                  ^-  (unit @)
+                  ?.  &((gte sur hsb) (lte sur hst))
+                    ~
+                  [~ (sub sur hsb)]
+              unic
+            ==
+            ;~  pfix
+              (jest '\\u')
+              %+  sear
+                |=  sur=@ux
+                  ^-  (unit @)
+                  ?.  &((gte sur lsb) (lte sur lst))
+                    ~
+                  [~ (sub sur lsb)]
+              unic
+            ==
+          ==
+        =*  tuf
+          =*  hsb  0xd800
+          =*  lst  0xdfff
+          %+  cook  tuft
+          ;~  pfix
+            (just 'u')
+            %+  sear
+              |=  sur=@ux
+                ^-  (unit @)
+                ?.  |((lth sur hsb) (gth sur lst))
+                  ~
+                (some sur)
+            unic
+          ==
+        ;~(pose doq fas bas loo sup tuf)
```

Comment:
Will do. May need further advice on optimization

Reply:
Done, please let me know if there are other optimizations to be made to `unic:de-json:html`.

Question related to my use of `=*`: I recall reading that `=*` is a macro, but according to [this PR in the urbit.org repo](https://github.com/urbit/urbit.org/pull/1254), it's actually considered a "deferred expression". Furthermore, based on the new docs for `=*` and @Fang- 's comment, it looks like the "alias" is re-evaluated every time it's used. Am I correctly understanding this, and if so, then should all of the usages of `=*` in `de-json` be removed?

Reply:
`=*` is a macro in the sense of C `#define` - more cleanly/strictly scoped, but otherwise identical to pasting the text later. I have not found any cases where it's preferable to `=/`, I think mostly it saves on stack space usage at the expense of also deferring expression *errors*.

Reply:
Gotcha - in that case, it sounds like I should leave it since I use `=*` to define `@ux` constants which will throw no errors.